File: Backend Code/src/main/java/com/LabResourceUtilizationPlatform/Service/InvoiceService.java
package com.LabResourceUtilizationPlatform.Service;

import com.LabResourceUtilizationPlatform.Dtos.Response.InvoiceResponse;
import com.LabResourceUtilizationPlatform.Entity.Booking;

import java.util.List;

public interface InvoiceService {

    InvoiceResponse getInvoice(Long bookingId);

    List<InvoiceResponse> getMyInvoices();

    void createInvoice(Booking booking);

    List<InvoiceResponse> getInstitutionInvoices();
}

File: Frontend Code/src/services/invoiceService.js
import api from "./api";

const invoiceService = {
  getInvoice: async (bookingId) => {
    const { data } = await api.get(`/invoices/${bookingId}`);
    return data;
  },

  getMyInvoices: async () => {
    const { data } = await api.get("/invoices/my");
    return data;
  },

  getInstitutionInvoices: async () => {
    const { data } = await api.get("/invoices/institution");
    return data;
  },
};

export default invoiceService;